packages/client/src/transaction/helpers/index.ts

```diff
@@ -0,0 +1,47 @@
+import { RelayerResultFragment, RelayErrorFragment } from '../../graphql/fragments.generated';
+import {
+  TransactionErrorFragment,
+  TransactionIndexedResultFragment,
+} from '../graphql/transaction.generated';
+
+export function isRelayerResult(
+  result: RelayerResultFragment | RelayErrorFragment,
+): result is RelayerResultFragment {
+  return 'txId' in result;
+}
+
+export function isRelayerError(
+  result: RelayerResultFragment | RelayErrorFragment,
+): result is RelayErrorFragment {
+  return 'reason' in result;
+}
+
+export function isTransactionIndexedResult(
+  result: TransactionIndexedResultFragment | TransactionErrorFragment,
+): result is TransactionIndexedResultFragment {
+  return 'indexed' in result;
+}
+
+export function isTransactionError(
+  result: TransactionIndexedResultFragment | TransactionErrorFragment,
+): result is TransactionErrorFragment {
+  return 'reason' in result;
+}
+
+export function getTxIdFromRelayerResult(
+  result: RelayerResultFragment | RelayErrorFragment,
+): string {
+  if (isRelayerResult(result)) {
+    return result.txId;
+  }
+  throw new Error(result.reason);
```

Comment:
Not sure if this error throw by these helpers is intelligible. What is the intent? The error is also not custom typed, what will happen to it? Will it be wrapped or something?

Reply:
exactly, I don't know, what should happen if the result is an instance of RelayErrorFragment? 

I have a need for a helper (ie. `extractTxIdFromRelayerResult`) to extract the values from return types. The full usage is

```ts
const profileCreateResult = await client.profile.create({ handle });

if (profileCreateResult.isSuccess()) {
  const value = profileCreateResult.value; // RelayerResultFragment | RelayErrorFragment
  
  if('txId' in value) {
    // tx to create a profile submitted with txId
  } else {
    // handle RelayErrorFragment
  }
} else {
  // profileCreateResult is a failure, means CredentialsExpiredError | NotAuthenticatedError
}
```

using helper
```ts
// profileCreateResult is a Result type, profileCreateResult.value is RelayerResultFragment | RelayErrorFragment
const txId = extractTxIdFromRelayerResult(profileCreateResult.unwrap()); // can we assume it is a success as client is authenticated?
  
if (!txId) {
  throw Error(`Profile creation failed`);
}
```

how would you model it?

Reply:
I just removed that helper, not needed anymore